### ISSUE TITLE
Traceback when [p]help+a command that doesn't exist

### ISSUE THREAD

**Issue by Gtoyos** [Cog-Creators/Red-DiscordBot#1746]
# Command bugs

#### Command name

$help

#### What cog is this command from?

Core?

#### What were you expecting to happen?

Nothing should happend. Or Red should send a message saying that command doesn't exist

#### What actually happened?

CommandInvokeError: Command raised an exception: AttributeError: 'coroutine' object has no attribute 'to_dict'
Error in command 'help'. Check your console or logs for details.

#### How can we reproduce this issue?

[p]help thisisnotacommand

----
### Traceback:
```
Exception in command 'help'
Traceback (most recent call last):
  File "/home/gtoyos/.local/lib/python3.6/site-packages/discord/ext/commands/core.py", line 62, in wrapped
    ret = yield from coro(*args, **kwargs)
  File "/home/gtoyos/.local/lib/python3.6/site-packages/redbot/core/help_formatter.py", line 330, in help
    await destination.send(embed=await ctx.bot.formatter.cmd_not_found(ctx, name))
  File "/home/gtoyos/.local/lib/python3.6/site-packages/discord/abc.py", line 742, in send
    embed = embed.to_dict()
AttributeError: 'coroutine' object has no attribute 'to_dict'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/gtoyos/.local/lib/python3.6/site-packages/discord/ext/commands/bot.py", line 886, in invoke
    yield from ctx.command.invoke(ctx)
  File "/home/gtoyos/.local/lib/python3.6/site-packages/discord/ext/commands/core.py", line 514, in invoke
    yield from injected(*ctx.args, **ctx.kwargs)
  File "/home/gtoyos/.local/lib/python3.6/site-packages/discord/ext/commands/core.py", line 71, in wrapped
    raise CommandInvokeError(e) from e
discord.ext.commands.errors.CommandInvokeError: Command raised an exception: AttributeError: 'coroutine' object has no attribute 'to_dict'
```

**Comment by Tobotimus** [comment]
Unfortunately I'm not able to reproduce this on the latest version. Could you please show the output of `redbot --version`?

**Comment by Gtoyos** [comment]
Version 3.0.0b15

**Comment by Tobotimus** [comment]
Thanks for bringing this issue forward!

Whilst this issue is present in the releases `3.0.0b15.post1` and `3.0.0b15.post2`, it has been fixed by #1707. Closing
